docs/config_options.rst

```diff
@@ -3,43 +3,143 @@ Configuration Options
+     - If tiles are cached in redis, the url or list of urls where the redis server is located.
+     - ``str | List[str]``
+     -
+     - ``"127.0.0.1:6379"``
+
+       .. _config_cache_redis_username:
+   * - ``cache_redis_username`` :ref:`🔗 <config_cache_redis_username>`
+     - A username for the redis server.
+     -
+     -
+     - ``None``
+
+       .. _config_cache_redis_password:
+   * - ``cache_redis_password`` :ref:`🔗 <config_cache_redis_password>`
+     - A password for the redis server.
+     -
+     -
+     - ``None``
+
+       .. _config_cache_tilesource_memory_portion:
+   * - ``cache_tilesource_memory_portion`` :ref:`🔗 <config_cache_tilesource_memory_portion>`
+     - Tilesources are cached on open so that subsequent accesses can be faster.  These use file handles and memory.  This limits the maximum based on a memory estimation and using no more than 1 / (``cache_tilesource_memory_portion``) of the available memory.
+     - ``int``
+     -
+     -
+
+       .. _config_cache_tilesource_maximum:
+   * - ``cache_tilesource_maximum`` :ref:`🔗 <config_cache_tilesource_maximum>`
+     - If this is non-zero, this further limits the number of tilesources than can be cached to this value.
+     -
+     -
+     -
+
+       .. _config_cache_sources:
+   * - ``cache_sources`` :ref:`🔗 <config_cache_sources>`
+     - If set to False, the default will be to not cache tile sources.  This has substantial performance penalties if sources are used multiple times, so should only be set in singular dynamic environments such as experimental notebooks.
+     -
+     -
+     -
+
+       .. _config_max_small_image_size:
+   * - ``max_small_image_size`` :ref:`🔗 <config_max_small_image_size>`
+     - The PIL tilesource is used for small images if they are no more than this many pixels along their maximum dimension.
+     - ``int``
+     -
+     -
+
+       .. _config_source_ignored_names:
+   * - ``source_bioformats_ignored_names``,
+       ``source_pil_ignored_names``,
+       ``source_vips_ignored_names`` :ref:`🔗 <config_source_ignored_names>`
+     - Some tile sources can read some files that are better read by other tilesources.  Since reading these files is suboptimal, these tile sources have a setting that, by default, ignores files without extensions or with particular extensions.
+     - ``str`` (regular expression)
+     -
+     - ``r'(^[!.]*|\.(jpg|jpeg|jpe|png|tif|tiff|ndpi))$'`` (for bioformats)
+
+       .. _config_all_sources_ignored_names:
+   * - ``all_sources_ignored_names`` :ref:`🔗 <config_all_sources_ignored_names>`
+     - If a file matches the regular expression in this setting, it will only be opened by sources that explicitly match the extension or mimetype.  Some formats are composed of multiple files that can be read as either a small image or as a large image depending on the source; this prohibits all sources that don't explicitly support the format.
+     -
+     -
+     -
```

Comment:
Default is `'(\.mrxs|\.vsi)$'`.  Typically, these should be formats that required multiple files to work but could also be opened as single files lacking details.